packages/babel-register/src/worker/cache.mts

```diff
@@ -0,0 +1,102 @@
+import path from "node:path";
+import fs from "node:fs";
+import os from "node:os";
+// @ts-expect-error no types
+import findCacheDir from "find-cache-dir";
+// @ts-expect-error no types
+import * as lmdb from "lmdb";
+
+let globalDisableCache = false;
+
+export default class Cache {
+  enabled = false;
+  cacheDir =
+    process.env.BABEL_CACHE_PATH ||
+    findCacheDir({ name: "@babel/register" }) ||
+    path.join(os.tmpdir() || os.homedir(), `.babel-register`);
+  db: lmdb.Database = null;
+  cache: Map<string, unknown> = null;
+
+  constructor() {
+    globalDisableCache = !!process.env.BABEL_DISABLE_CACHE;
+
+    if (fs.existsSync(this.cacheDir)) {
+      if (fs.statSync(this.cacheDir).isFile()) {
+        console.warn(
+          `Cache directory ${this.cacheDir} is a file, not a directory.`,
+        );
+        globalDisableCache = true;
+      }
+      return;
+    }
+    try {
+      fs.mkdirSync(this.cacheDir, { recursive: true });
+    } catch (_) {
+      console.warn(`Failed to create cache directory ${this.cacheDir}`);
+      globalDisableCache = true;
+      return;
+    }
+    try {
+      fs.accessSync(this.cacheDir, fs.constants.W_OK | fs.constants.R_OK);
+    } catch (_) {
+      console.warn(
+        `Cache directory ${this.cacheDir} is not writable or readable.`,
+      );
+      globalDisableCache = true;
+    }
+  }
+
+  enable() {
+    if (globalDisableCache) return;
+    this.cache = new Map();
+    this.enabled = true;
+
+    if (!this.db) {
+      this.db = lmdb.open({
+        path: path.join(this.cacheDir, "cache.lmdb"),
+        compression: true,
+        noSync: true,
+      });
+
+      let createdAt = this.db.get("created_at");
+      // If the cache is older than 30 days, clear it
```

Comment:
If I compile something every day for 31st day, on the 31st day will it delate all the cache or the data from the 1t day?